java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessStatement.java

```diff
@@ -374,8 +376,10 @@ public ResultSet getGeneratedKeys() throws SQLException {
         String[][] data = null;
 
         if (this.generatedId > 0) {
-            data = new String[1][1];
-            data[0][0] = String.valueOf(this.generatedId);
+            data = new String[(int) this.resultCount][1];
```

Comment:
I presume you're mimicking the mysql JDBC driver? Can you add a comment that says so?